Packages/com.unity.inputsystem/InputSystem/Controls/IOptimizedControl.cs

```diff
@@ -0,0 +1,34 @@
+﻿namespace UnityEngine.InputSystem.Controls
+{
+    public enum OptimizedControlDataType
+    {
+        // If control state cannot be casted to any of types below due to various reasons
+        NotOptimized = 0,
+
+        // Bool value as 8 bit byte, byte aligned, where 0 means false and any other value means true
+        BoolAsByte,
+        
+        // Integer as 32 bit, byte aligned
+        Int32,
+        
+        // Float as 32 bit, byte aligned
+        Float,
+        
+        // Vector2 as 2 consecutive 32 bit floats, in order x,y, byte aligned
+        Vector2,
+
+        // Vector3 as 3 consecutive 32 bit floats, in order x,y,z, byte aligned
+        Vector3,
+
+        // Quaternion as 4 consecutive 32 bit floats, in order of x,y,z,w, byte aligned
+        Quaternion,
+
+        // PoseState
+        Pose
+    }
+    
+    internal interface IOptimizedControl
+    {
+        public OptimizedControlDataType CalculateOptimizedControlDataType();
```

Comment:
Ok that's not gonna fly:
- Let's say we have a custom control that derives from `AxisControl` and does custom processing in `ReadUnprocessedValueFromState`
- We make `Vector3Control` of such controls
- Now `Vector3Control` has no way to check if the nested controls are optimized or not, because interface is already implemented on `AxisControl` level and can't be implemented again in derived controls.

Suggestion: use virtual method on `InputControl` instead

Reply:
Fixed